src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -513,8 +513,8 @@ private bool IsLive(Project project, Compilation compilation, DiagnosticData dia
                 // REVIEW: current design is that we special case compiler analyzer case and we accept only document level
                 //         diagnostic as live. otherwise, we let them be build errors. we changed compiler analyzer accordingly as well
                 //         so that it doesn't report project level diagnostic as live errors.
-                if (_owner._diagnosticService.AnalyzerInfoCache.IsCompilerDiagnostic(project.Language, diagnosticData) &&
-                    !IsDocumentLevelDiagnostic(diagnosticData))
+                if (!IsDocumentLevelDiagnostic(diagnosticData) &&
+                    diagnosticData.CustomTags.Contains(WellKnownDiagnosticTags.Compiler))
```

Comment:
I am not sure if the diagnostics generated from build output have valid custom tags. Regardless of whether they do or not, we do need a test for this case.

Reply:
Current implementation of `IsCompilerDiagnostic` seems expensive as it force creates diagnostic descriptors for all compiler diagnostics, so it is good that you are removing it. I think it might be reasonable to just check diagnostic ID format `CSxxxx` and `BCxxxxx`.

Reply:
Yes, I was wondering if we could just check CS and BC prefixes. Would that be ok?

Reply:
Actually, the external diagnostics use the right tag: 
https://github.com/dotnet/roslyn/blob/master/src/VisualStudio/Core/Def/Implementation/TaskList/ProjectExternalErrorReporter.cs#L297
So this should work fine.

Reply:
Ah that is good to know! Thanks.